include/crocoddyl/multibody/numdiff/cost.hpp

```diff
@@ -0,0 +1,189 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2020, LAAS-CNRS, New York University,
+//                          Max Planck Gesellschaft
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MULTIBODY_NUMDIFF_COST_HPP_
+#define CROCODDYL_MULTIBODY_NUMDIFF_COST_HPP_
+
+#include <boost/function.hpp>
+#include "crocoddyl/multibody/cost-base.hpp"
+#include "pinocchio/algorithm/kinematics.hpp"
+#include "pinocchio/algorithm/jacobian.hpp"
+#include "pinocchio/algorithm/frames.hpp"
+#include "pinocchio/algorithm/center-of-mass.hpp"
+#include "pinocchio/algorithm/centroidal.hpp"
+#include "pinocchio/algorithm/centroidal-derivatives.hpp"
+
+namespace crocoddyl {
+
+struct CostDataNumDiff;  // forward declaration
+
+// Simple renaming that ease the code writing.
+typedef boost::function<void(const Eigen::VectorXd&)> ReevaluationFunction;
+
+/**
+ * @brief Compute all the pinocchio data needed for the numerical
+ * differentation. We use the address of the object to avoid a copy from the
+ * "boost::bind".
+ *
+ * @param model is the rigid body robot model.
+ * @param data contains the results of the computations.
+ * @param x is the state vector.
+ */
+void updateAllPinocchio(pinocchio::Model* const model, pinocchio::Data* data, const Eigen::VectorXd& x) {
```

Comment:
I believe we should either move this function to test_costs.cpp file, or a common used file for unittest code (if we need to reuse it in other tests).